examples/frameworks/shiny/main.py

```diff
@@ -0,0 +1,86 @@
+# /// script
+# requires-python = ">=3.12"
+# dependencies = [
+#     "shiny",
+#     "marimo",
+#     "vega-datasets==0.9.0",
+# ]
+# ///
+from shiny import App, render, ui
+from htmltools import HTML
+import marimo
+import os
+from starlette.middleware.wsgi import WSGIMiddleware
+from starlette.applications import Starlette
+from starlette.routing import Mount, Route
+from starlette.responses import RedirectResponse
+
+
+ui_dir = os.path.join(os.path.dirname(__file__), "..", "..", "ui")
+templates_dir = os.path.join(os.path.dirname(__file__), "templates")
+
+marimo_app = marimo.create_asgi_app()
+app_names: list[str] = []
+
+for filename in sorted(os.listdir(ui_dir)):
+    if filename.endswith(".py"):
+        app_name = os.path.splitext(filename)[0]
+        app_path = os.path.join(ui_dir, filename)
+        marimo_app = marimo_app.with_app(path=f"/{app_name}", root=app_path)
+        app_names.append(app_name)
+
+shiny_ui = ui.page_fluid(ui.output_ui("_html"))
+
+
+def shiny_server(input, output, session):
+    @render.text
+    def _html():
+        subdiv = []
+        for name in app_names:
+            subdiv.append(
+                f"""
+                <div class="bg-base-200 rounded-lg shadow-md hover:shadow-xl 
+                    transition-shadow duration-300 w-64">
+                  <a href="/{name}" target="_blank" class="block w-full h-full">
+                    <div class="p-4">
+                      <h2 class="text-xl font-semibold mb-2">{name}</h2>
+                      <p class="text-sm text-gray-500">Click to open app</p>
+                    </div>
+                  </a>
+                </div>
+                """
+            )
+        head = """
+<head>
+    <script src="https://cdn.tailwindcss.com" />
+    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/daisyui@4.11.1/dist/full.min.css"
+    <title>Marimo apps in Shiny</title>
+</head>
+        """
```

Comment:
To ensure this JS/CSS gets included on the page only once, put it in the UI instead:

```python
shiny_ui = ui.page_fluid(
    ui.head_content(
        ui.tags.script(src="https://cdn.tailwindcss.com"),
        ui.tags.link(href="https://cdn.jsdelivr.net/npm/daisyui@4.11.1/dist/full.min.css", rel="stylesheet"),
    ),
    ui.output_ui("_html"),
    title="Marimo apps in Shiny"
)
```